File: Frontend/cosmetics/src/types/product.ts

export interface Product {
    id?: string;
    brand?: string;
    category?: string;
    description?: string;
    discount?: number;
    name: string;
    price: number;
    slug?: string;
    stock?: number;
    image?: string;
    discountedPrice?: number;
}


export interface ProductCardProps {
    product: Product;
    onAddToOrder: (product: Product) => void;
}

File: Backend/product-service/src/main/java/com/example/product_service/entity/Product.java
package com.example.product_service.entity;

import jakarta.persistence.*;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;

    String name;

    String image;

    String description;

    String brand;

    String category;

    Long price;

    Double discount;

    Integer stock;

    String slug;
}
